lib/Optimizer/Transforms/PySynthCallableBlockArgs.cpp

```diff
@@ -97,10 +132,13 @@ class UpdateQuakeApplyOp : public OpConversionPattern<quake::ApplyOp> {
 class PySynthCallableBlockArgs
     : public cudaq::opt::PySynthCallableBlockArgsBase<
           PySynthCallableBlockArgs> {
+private:
+  bool removeBlockArg = false;
```

Comment:
Q: callables are always going to be entry-block arguments, so if we're removing them from there, we'll also have to update the FuncOp signature, right?

Reply:
correct, which is done with the op.eraseArguments() call. These changes build off what was there previously for the python ast bridge / py_alt_launch_kernel. There the funcop update is not required, so I made this an optional flag to retain backwards compatibility